### PR TITLE
Message queue overlay

### PR DESCRIPTION
**Pull request by RipleyTom**
Adds a message queue overlay to add messages to a display queue.

Atm only used to make success/failure to login on RPCN easily visible to users:
image

### CODE REVIEW

File: rpcs3/Emu/RSX/Overlays/overlay_message.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+			{
+				return {};
+			}
+
+			return m_text.get_compiled();
+		}
+
+		void message_item::update(usz index, u64 time)
+		{
+			m_processed = true;
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/Overlays/overlay_message.cpp:37]: Logically this should be the final line in this function

File: rpcs3/Emu/RSX/Overlays/overlay_message.cpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			{
+				m_cur_pos = index;
+				m_text.set_pos(10, index * 18);
+			}
+
+			const u64 diff = m_expiration_time - time;
+
+			if (diff < 2'000'000)
+			{
+				m_text.fore_color.a = (diff / 2'000'000.f);
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/Overlays/overlay_message.cpp:49]: I think for future compatibility this should use the animation_color_interpolate class

Comment (kd-11) [reply rpcs3/Emu/RSX/Overlays/overlay_message.cpp:49]: Seconded. We have animation classes for overlays that are fairly easy to use.

File: rpcs3/Emu/RSX/Overlays/overlay_message.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+		void message_item::update(usz index, u64 time)
+		{
+			if (m_cur_pos != index)
+			{
+				m_cur_pos = index;
+				m_text.set_pos(10, index * 18);
+			}
+
+			if ((m_expiration_time - time) < 2'000'000)
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/Overlays/overlay_message.cpp:50]:
```suggestion
			if (fade_animation.active)
```

Comment (RipleyTom) [reply rpcs3/Emu/RSX/Overlays/overlay_message.cpp:50]: I only want to apply it when there is less than 2 seconds left...

**Pull request merged**